lostPatches.sh - find missing patches that are in one branch but not …

robstryker opened the pull request:
…in another

robstryker commented:
So far i've just cleaned it up, and it's just a shell script and nothing more.  I'm not really sure how to get this to be on a level where it can send emails or open jiras :(

nickboldt commented on util/findLostPatches.py:
```diff
@@ -0,0 +1,165 @@
+#    rootJBIDE.delete(deleteSubtasks=True)
+#    rootJBDS.delete(deleteSubtasks=True)
+
+# For sample usage, see createTaskJIRAs.py.examples.txt
```
Maybe want to remove that?

nickboldt commented on util/findLostPatches.py:
```diff
@@ -0,0 +1,165 @@
+## map from descriptive name to list of JBIDE and/or JBDS components.
+JBT_components = {
+
+# active projects
```
We should figure out how to externalize this to another file so that this script and the ones in jbdevstudio-ci/bin/*.py can reuse the mappings w/o having to update in multiple places.

robstryker replied:
I agree... but I did need to change or remove some entries because my logic for changing it into a repo url didn't work.  ie,  jbosstools-$name.lower()  

So this must be remembered.

nickboldt commented on util/findLostPatches.py:
```diff
@@ -0,0 +1,165 @@
+def quote(x):
+    return '"' + x + '"'
+
+
```
Many      spaces     here.

robstryker replied:
Copied from your original class ;)   I just use a trim() to get around it :D

nickboldt commented on util/findLostPatchesOneRepository.sh:
```diff
@@ -0,0 +1,121 @@
+#   Find any missing commits that are IN a given branch 
+#   but are NOT in the 2nd branch. 
+#  
+#   These missing commits are found via a concept
```
"via a concept called patch-id" sounds a bit high-fallutin'. Why not just "via patch-id" ? Ya'll gettin' fancy?

robstryker replied:
some of us just fancy by nature. Must be my New York tax-hiking, government-expanding, latte-drinking, sushi-eating, Volvo-driving, New York Times-reading, body-piercing, Hollywood-loving, left-wing freak show values.

nickboldt replied:
You don't drive a Volvo and I'm pretty sure you don't have any piercings. So... quit your posturing, yo. :P

nickboldt commented on util/findLostPatchesOneRepository.sh:
```diff
@@ -0,0 +1,121 @@
+        MASTERPATCHID+=($TMPPATCHID)
+   done
+   
+
```
again with the many empty lines.

robstryker replied:
Nothing wrong with empty lines.  spaces... on the other hand...

nickboldt commented on util/findLostPatchesOneRepository.sh:
```diff
@@ -0,0 +1,121 @@
+   if [ "x$BRANCH1" = "x" ]
+     then
+       echo "Usage: findLostPatchesOneRepository.sh jbosstools-4.3.x [master]";
+       exit 1;
```
how does this call the python script to create JIRAs?

nickboldt commented:
Having added JIRA generation, is this now ready to merge? Thx for adding example of how to use.

robstryker commented on util/findLostPatches.py:
```diff
@@ -0,0 +1,176 @@
+    os.system("git clone " + githubrepo)
+    os.chdir(workingdir + "/" + subfoldername.strip())
+    #  TODO externalize the branch here
+    p = Popen(['bash', '../../findLostPatchesOneRepository.sh', 'jbosstools-4.3.x'], stdout=PIPE, stderr=PIPE, stdin=PIPE)
```
This is the last piece that I believe needs 100% to be externalized, but I'm not quite sure where / how to put it.  @nickboldt  can you help here?

robstryker commented:
I made some changes to one of the scripts. Now, we get a lot more information.  First, i've added ability to find possible matching commits based on commit message. Second, I've shown a diff of the two commits so that reviewers can more clearly see what is different between the two. 

Take for example the two following patches:
   https://github.com/jbosstools/jbosstools-openshift/commit/9825476239f56ce6ee2ea55fd5a6aa5232bc51a8
   https://github.com/jbosstools/jbosstools-openshift/commit/fcdfc9a8da1d9481bb7f9734577641c924ef8f4c

A reviewer might see them and think they are identical because they're not paying attention. I now added a link to a pastebin where the diff of the two patches can be seen:  http://sprunge.us/SJee

See also https://issues.jboss.org/browse/JBIDE-22209  to see the new output in action.

maxandersen commented on util/findLostPatches.py:
```diff
@@ -0,0 +1,178 @@
+from jira.client import JIRA
```
the python scripts naming convention is lower case, so this should be findlostpatches.py

nickboldt commented:
PR refactored (lowercase filenames) and merged to 4.4.x branch.
